orbit/pkg/table/mdm/mdm_windows.go

```diff
@@ -0,0 +1,402 @@
+
+	procIsDeviceRegisteredWithManagement *windows.LazyProc = modmdmregistration.NewProc("IsDeviceRegisteredWithManagement")
+	procSendSyncMLcommand                *windows.LazyProc = modmdmlocalmgmt.NewProc("ApplyLocalManagementSyncML")
+
+	// Synchronization mutex
+	mu sync.Mutex
+
+	// SMBIOS UUID SHA256 hash generation executor
+	uuidHashGeneration sync.Once
+
+	// SHA256 hash of device SMBIOS UUID
+	uuidHash []byte
+)
+
+const (
+	maxBufSize            = 2048 // Max Unicode Length for UPN - https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-adls/63f5e067-d1b3-4e6e-9e53-a92953b6005b
+	flagsRegistryLocation = `SYSTEM\CurrentControlSet\Services\embeddedmode\Parameters`
+	flagsRegistryName     = `Flags`
+	mdmMutexName          = "__MDM_LOCAL_MANAGEMENT_NAMED_MUTEX__"
+)
+
+// Columns is the schema of the table.
+func Columns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.TextColumn("enrollment_status"),
+		table.TextColumn("enrolled_user"),
+		table.TextColumn("mdm_command_input"),
+		table.TextColumn("mdm_command_output"),
+	}
+}
+
+// Generate is called to return the results for the table at query time.
+// Constraints for generating can be retrieved from the queryContext.
+func Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	// grabbing command input query if present
+	var inputCmd string
+
+	// checking if 'mdm_command_input' is in the where clause
+	if constraintList, present := queryContext.Constraints["mdm_command_input"]; present {
+		for _, constraint := range constraintList.Constraints {
+			if constraint.Operator == table.OperatorEquals {
+				inputCmd = constraint.Expression
+				log.Debug().Msgf("mdm_bridge input command request:\n%s", inputCmd)
+			}
+		}
+	}
+
+	// getting MDM enrollment status
+	isHostMDMenrolled, enrollmentURI, err := getEnrollmentInfo()
+	if err != nil {
+		return nil, fmt.Errorf("there was a problem getting enrollment info: %s ", err)
+	}
+
+	// table returns immediately if host is not MDM enrolled
+	if isHostMDMenrolled == 0 {
+		return []map[string]string{
+			{
+				"enrollment_status":  "device_not_enrolled",
+				"enrolled_user":      "",
+				"mdm_command_input":  inputCmd,
+				"mdm_command_output": "",
+			},
+		}, nil
+	}
+
+	// Executing the input MDM command if it was present, otherwise just return with MDM enrolled info
+	if len(inputCmd) > 0 {
+
+		// performs the actual MDM cmd execution against the OS MDM stack
+		outputCmd, err := executeMDMcommand(inputCmd)
+		if err != nil {
+			return nil, fmt.Errorf("mdm command execution: %s ", err)
+		}
+
+		log.Debug().Msgf("mdm_bridge output command response:\n%s", outputCmd)
+
+		return []map[string]string{
+			{
+				"enrollment_status":  "enrolled_device",
+				"enrolled_user":      enrollmentURI,
+				"mdm_command_input":  inputCmd,
+				"mdm_command_output": outputCmd,
+			},
+		}, nil
+	} else {
+		return []map[string]string{
```

Comment:
Nothing critical, but mentioning it since some linters raise a warning about that (although not enabled in our repo) and is somewhat related to [this style guide recommendation](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow): when an `if` clause ends unconditionally with a `return`, you should remove the `else` and unindent the rest of the code. The rationale being that if it enters the `if`, it won't exit it and so there's no need to wrap the rest of the code in an `else`, and less-nested code is usually a bit more readable.

I see this in a couple other places in the PR too.

Reply:
Good comment, I've just improved this through the code. I've also relaxed the logic a bit to allow MDM commands to be processed even if the device is not MDM enrolled.